Scripts/Mac/BuildInstaller.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+CONFIGURATION=$1
+if [ -z $CONFIGURATION ]; then
+  CONFIGURATION=Debug
+fi
+
+PACKAGEVERSION=$2
+if [ -z $PACKAGEVERSION ]; then
+  PACKAGEVERSION="1.0.0.1"
+fi
+
+SCRIPTDIR="$(dirname ${BASH_SOURCE[0]})"
+
+# convert to an absolute path because it is required by `dotnet publish`
+pushd $SCRIPTDIR
+SCRIPTDIR="$(pwd)"
+popd
+
+SRCDIR=$SCRIPTDIR/../..
+ROOTDIR=$SRCDIR/..
+BUILDOUTPUT=$ROOTDIR/BuildOutput
+PUBLISHDIR=$ROOTDIR/Publish
+INSTALLROOT=$ROOTDIR/InstallRoot
+VFSFORGITDESTINATION="usr/local/vfsforgit"
+INSTALLERPACKAGENAME="VFSForGit.$PACKAGEVERSION"
+
+echo "SRCDIR: ${SRCDIR}"
+echo "ROOTDIR: ${ROOTDIR}"
+echo "BUILDOUTPUT: ${BUILDOUTPUT}"
+echo "PUBLISHDIR: ${PUBLISHDIR}"
+
+function CheckBuildIsAvailable()
+{
+	if [ ! -d $BUILDOUTPUT ] || [ ! -d $PUBLISHDIR ]; then
+  		echo "Could not find VFSForGit Build to package."
+  		exit 1
+	fi
+}
+
+function SetPermissions()
+{
+	chmodCommand="chmod -R 755 \"${INSTALLROOT}\""
+	echo $chmodCommand
+	eval $chmodCommand || exit 1
+}
+
+function CreateInstallerRoot()
```

Comment:
We need to run this code as part of the installation of the package, not as part of the script that creates the package. You can pass pre/postinstall scripts to pkgbuild and they'll be added to the package and ran at the correct times.

Reply:
Ameen and I talked offline, this pattern works fine, it's just different than what I expected :)